lib/api/core.dart

```diff
@@ -214,9 +215,39 @@ Map<String, String> authHeader({required String email, required String apiKey})
 }
 
 Map<String, String> userAgentHeader() {
+  final deviceInfo = ZulipBinding.instance.deviceInfo;
+  final PackageInfo(
+    version: appVersion,
+    buildNumber: appBuildNumber) = ZulipBinding.instance.packageInfo;
+
+  final (osName, osVariant) = switch (deviceInfo) {
+    AndroidDeviceInfo(
+      :var sdkInt)        => ('Android', '$sdkInt'), // "34"
+    IosDeviceInfo(
+      :var systemVersion) => ('iOS', systemVersion), // "17.4"
+    MacOsDeviceInfo(
+      :var majorVersion,
+      :var minorVersion,
+      :var patchVersion)  => ('macOS', '$majorVersion.$minorVersion.$patchVersion'), // "14.5.0"
+    WindowsDeviceInfo(
+      :var majorVersion,
+      :var minorVersion,
+      :var buildNumber)   => ('Windows', '$majorVersion.$minorVersion $buildNumber'), // "10.0 22631" (actually means Windows 11, 23H2)
```

Comment:
Unlike other OSes - Windows doesn't seem to provide a way to get a recognizable result from it's provided values, we have to parse the number ourselves. See https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/ns-wdm-_osversioninfoexw#remarks
For example, [here's how chromium parses it](https://source.chromium.org/chromium/chromium/src/+/main:base/win/windows_version.cc;l=371-470;drc=93b31639a1e56424686adc90b1de30aaec981dfc) (BSD 3-clause licensed).

I felt like rather than doing that guessing work/parsing in the client, we should report concrete numbers, what are you thoughts?

Reply:
For Windows I'd prioritize keeping to a minimum the effort and code we spend. So I'd be perfectly happy with reporting no OS version at all, just saying "Windows" with no further details, or with reporting some version string that's easy to get at.

That goes for all three desktop platforms, really — see https://github.com/zulip/zulip-flutter#desktop-support — but especially for Windows, because I'm not sure anyone has even ever run the app there.

As far as I'm aware, the first time any of our code ran on Windows was our test suite when I sent a PR to add us to flutter/tests; see #700, then `flutter test` failed with a mysterious other error, so before merge I skipped Windows in our flutter/tests entry. That means even our tests have been run on Windows a grand total of twice that I'm aware of.